### PR TITLE
Fix default argument for backend in compile_pymc_model

### PR DESCRIPTION
**Pull request by fonnesbeck**
The argument handling for `backend` in `compile_pymc_model` implies that it should have `None` as a default value. This changes the default argument from "numba" to None.

### CODE REVIEW

File: python/nutpie/compile_pymc.py [hunk 1]
```diff
@@ -354,7 +354,7 @@ def expand(x, **shared):
 def compile_pymc_model(
     model: "pm.Model",
     *,
-    backend: Literal["numba", "jax"] = "numba",
+    backend: Literal["numba", "jax"] | None = None,
```
Comment (ricardoV94) [inline python/nutpie/compile_pymc.py:357]: I don't follow, why this change. The docstrings don't mention anything about `None`. The code checks for it below but it seems having numba as the explicit default is more readable?

Comment (fonnesbeck) [reply python/nutpie/compile_pymc.py:357]: The code implies that None is expected. We should either have both `backend` and `gradient_backend` specified with their None or the actual default; having them being handled differently is odd.

Comment (fonnesbeck) [reply python/nutpie/compile_pymc.py:357]: I think None is less bug prone because upstream defaults don't have to know what the default actually is (which can cause bugs if it ever changes).

Comment (ricardoV94) [reply python/nutpie/compile_pymc.py:357]: I'm on you with this one, but if you followed that logic to the end all arguments would have to accept None :)
